www/js/controlHelper.ts

```diff
@@ -0,0 +1,136 @@
+        logDebug(`fileEntry ${fileEntry.nativeURL} is file? ${fileEntry.isFile.toString()}`);
+        fileEntry.file(function(file) {
+          const reader = new FileReader();
+
+          reader.onloadend = function() {
+            const readResult = this.result as string;
+            logDebug(`Successfull file read with ${readResult.length} characters`);
+            const dataArray = JSON.parse(readResult);
+            logDebug(`Successfully read resultList of size ${dataArray.length}`);
+            let attachFile = fileEntry.nativeURL;
+              if (window['device'].platform === "android")
+                attachFile = "app://cache/" + fileName;
+              if (window['device'].platform === "ios")
+                alert(i18next.t("email-service.email-account-mail-app"));
+              const email = {
+                'files': [attachFile],
+                'message': i18next.t("email-service.email-data.body-data-consists-of-list-of-entries"),
+                'subject': i18next.t("email-service.email-data.subject-data-dump-from-to", {start: startTimeString ,end: endTimeString}),
+              }
+              window['plugins'].socialsharing.shareWithOptions(email, function (result) {
+                logDebug(`Share Completed? ${result.completed}`); // On Android, most likely returns false
+                logDebug(`Shared to app:  ${result.app}`);
+                resolve();
+              }, function (msg) {
+                logDebug(`Sharing failed with message ${msg}`);
+              });
+          }
+          reader.readAsText(file);
+        }, function(error) {
+          displayError(error, "Error while downloading JSON dump");
+          reject(error);
+        })                        
+      });
+    });
+    });
+}};
+
+/**
+ * getMyData fetches timeline data for a given day, and then gives the user a prompt to share the data
+ * @param startTs initial timestamp of the timeline to be fetched.
+ */
+export const getMyData = function(startTs: Date) {
+    // We are only retrieving data for a single day to avoid
+    // running out of memory on the phone
+    const startTime = DateTime.fromJSDate(startTs);
+    const endTime = startTime.endOf("day");
+    const startTimeString = startTime.toFormat("yyyy'-'MM'-'dd");
+    const endTimeString = endTime.toFormat("yyyy'-'MM'-'dd");
+
+    const dumpFile = startTimeString + "."
+      + endTimeString
+      + ".timeline";
+      alert(`Going to retrieve data to ${dumpFile}`);
+
+    // Simulate old conversion to get correct UnixInteger for endMoment data
+    const getUnixNum = (dateData: DateTime) => {
+      const tempDate = dateData.toFormat("dd MMM yyyy");
+      return DateTime.fromFormat(tempDate, "dd MMM yyyy").toUnixInteger();
+    };
+
+    const writeDumpFile = createWriteFile(dumpFile);
+    const shareData = createShareData(dumpFile, startTimeString, endTimeString);
+
+    getRawEntries(null, getUnixNum(startTime), startTime.toUnixInteger())
```

Comment:
you get a `startTs` as input into the function.
I'm not sure about why you need to convert back and forth to `startTime` and use `getUnixNum`
Also, why are both the start and end, the `startTime`?
Again, have you checked on whether you actually get a file with data?

Reply:
Sorry for the confusion - yes, I've checked with a few different dates/timelines, on iOS and Android! I'll go ahead and do some re-tests after fixing these revisions.  

As for the reason we need `getUnixNum`, Thank you for asking!! I honestly struggled for a while to pick this problem apart.  Revisiting the issue, I'm not actually sure we need it.  To the best of my understanding, here's the problem:

- Before switching to ReactNativePaper, we used an Angular date-picker, and converted the input date to a `moment()` object (`dd MMM yyyy`).  After we switched to reactNativePaper, we started using a different time-string format (`dd-mm-yyyy`), which caused some problems (see: [issue](https://github.com/e-mission/e-mission-docs/issues/994#issuecomment-1745866273)).
- Different strings are converted to different Unix numbers differently: For example, If I have the following code...
```
    console.log(`\n\n==> Start Time : ${startTime} 
                 \nUnixNum: ${startTime.toUnixInteger()} \nConversionTime: ${getUnixNum(startTime)}`);
    console.log(`\n\n==> End Time : ${endTime} 
                 \nUnixNum: ${endTime.toUnixInteger()} \nConversionTime: ${getUnixNum(endTime)}`);
```
The resulting Unix values are different:
```
==> Start Time : 2023-04-14T23:59:59.000-07:00   
[phonegap] UnixNum: 1681541999 
[phonegap] ConversionTime: 1681455600

==> End Time : 2023-04-14T23:59:59.999-07:00 
UnixNum: 1681541999 
ConversionTime: 1681455600
```
- As the UnixNumber and Conversion Times show, using the "conversion" method simply rounds the time to the nearest hour.  I think this is what I was trying to figure out in [this](https://github.com/e-mission/e-mission-phone/pull/1052#issuecomment-1749381370) comment.  The issue may not be the date conversion, but rather, the fact that Luxon performs the "endOfDay" function differently than Moment.js. 

All of this to say - going back to the numbers, it appears I was using the old conversion as a _really_ hacky way to get the correct second precision, and just wasn't aware that was the issue.  This method works for all timelines I've tested, but now that I have more clarity on the issue, I'll go ahead and figure out a proper fix! ( I remember reading on a related issue that someone else was having problems with the moment/luxon precision - and, in that code, they used a "timezone" flag to ensure the correct "endOfDay" value.  I can't find the thread again, but I'll start by looking into this!)

Reply:
Update: I was able to refactor the code so it doesn't need the 'getUnixTime' hack! I'll be adding a commit with a few of these revisions, plus the updated code, here.
As for why `startTs` and `endTs` both come from the startTs - since we can only fetch one day at at time, we only need start value, and can figure out the Unix Integers (used to fetch the data) from that single date.  That being said, the name "startTs" is a bit misleading, so I'll go ahead and rename that!